src/target/target.c

```diff
@@ -64,19 +79,20 @@ target_s *target_new(void)
 	} else
 		target_list = t;
 
-	t->attach = (void *)nop_function;
-	t->detach = (void *)nop_function;
-	t->mem_read = (void *)nop_function;
-	t->mem_write = (void *)nop_function;
-	t->reg_read = (void *)nop_function;
-	t->reg_write = (void *)nop_function;
-	t->regs_read = (void *)nop_function;
-	t->regs_write = (void *)nop_function;
-	t->reset = (void *)nop_function;
-	t->halt_request = (void *)nop_function;
-	t->halt_poll = (void *)nop_function;
-	t->halt_resume = (void *)nop_function;
-	t->check_error = (void *)false_function;
+	t->attach = target_nop_true;
+	t->detach = target_nop_void;
+	t->mem_read = target_nop_void;
+	t->mem_write = target_nop_void;
+	t->reg_read = target_nop_zero;
+	t->reg_write = target_nop_zero;
+	t->regs_read = target_nop_void;
+	t->regs_write = target_nop_void;
+	t->reset = target_nop_void;
+	t->halt_request = target_nop_void;
+	t->halt_poll = target_nop_poll;
+	/* XXX: For some reason, if the cast is omitted, this one line generates a type mismatch error. */
```

Comment:
I suspect it's an ABI thing. Functions with undefined argument lists can be passed to function pointers taking integral arguments, but not `bool` or `float`, so I'm guessing it warns when the argument has a type that's not necessarily compatible with the integral ABI.

Example: https://godbolt.org/z/9bf8rG9sv

While we're at it, it might be worthwhile to just make default functions that's taking the actual argument types.

Reply:
Good point regarding bool/float.. thank you for giving us a working reason as to why we saw the behaviour we did. Regarding actual types, we thought about that and for the amount of extra code that generates, it would be better to entirely get rid of these default implementations and instead do NULL checks in the target API if we go there.. this already increased firmware size a fair bit. At least with argument-less functions the stack is properly dealt with which is why this current implementation works.

Reply:
We've replaced the approach with proper NULL checking and fixed some extra bugs in the target memory functions in the process :)